lib/identity_cache/query_api.rb

```diff
@@ -122,7 +122,12 @@ def check_association_scope(association_name)
 
       def record_from_coder(coder) #:nodoc:
         if coder
-          klass = coder[:class]
+          klass =
+            if coder[:class].is_a?(String)
+              coder[:class].constantize
+            else
+              coder[:class]
```

Comment:
Changing how we write data to memcache means we would have to bump the CACHE_VERSION.  We use that as part of the cache key, so once that is bumped we wouldn't have to support the deprecated code path.  It would also mean the cache would effectively get flushed as a result.